### PR TITLE
[draft] Misc stuff when porting ze

### PR DESCRIPTION
**Pull request by TApplencourt**

### CODE REVIEW

File: bin/metababel [hunk 1]
```diff
@@ -229,7 +230,13 @@ def parse_argv
 end
 
 def get_dispatch_type(em, default_clock_class, context)
-  dispatchers = em.domain ? OpenStruct.new(context).instance_eval(em.domain) : context['all']
+
+  begin
+    dispatchers = em.domain ? OpenStruct.new(context).instance_eval(em.domain) : context['all']
+  rescue StandardError
+    raise "Be sure your domain '#{em.domain}' is a valid ruby expresion and all set_id have been defined. Current defined set_id: #{context.keys}"
```
Comment (DonAurelio) [inline bin/metababel:237]:
```suggestion
    raise "Please ensure the domain '#{em.domain}' uses valid ruby expressions and all set_id have been previously defined. Current defined set_ids: #{context.keys}."
```

File: template/source.c.erb [hunk 2]
```diff
@@ -97,8 +103,7 @@ source_initialize(bt_self_component_source *self_component_source,
   /* Register User Callbacks */
   btx_register_usr_callbacks((void *)common_data);
   /* Call initialize_processing*/
-  btx_call_callbacks_initialize_component((void *)common_data,
-                                          &common_data->usr_data);
+  btx_call_callbacks_initialize_component(common_data, &common_data->usr_data);
```
Comment (DonAurelio) [inline template/source.c.erb:106]: Why was the (void *) cast removed for common_data?. Same in sink and filter.

**Review by DonAurelio — COMMENTED** [review]
Look good Thomas, just one question

File: template/source.c.erb (hunk 2 above)
Comment (TApplencourt) [reply template/source.c.erb:106]: Not required, `btx_call_callbacks_initialize_component` doesn't take a void pointer. Not sure why

**Pull request merged**